misk-aws/src/main/kotlin/misk/dynamodb/DyVersionedEntity.kt

```diff
@@ -0,0 +1,16 @@
+package misk.dynamodb
+
+/**
+ * This version field is used for optimistic locking.
+ */
+interface DyEntity {
```

Comment:
These are now interfaces, but I left the needed annotations in comments to help people figure out how to use them.